pandas/tslib.pyx

```diff
@@ -416,12 +416,21 @@ class Timestamp(_Timestamp):
         cdef object result, value
 
         from pandas.tseries.frequencies import to_offset
-        unit = to_offset(freq).nanos
+        offset = to_offset(freq)
+        unit = offset.nanos
+        time_unit = offset.rule_code
         if self.tz is not None:
             value = self.tz_localize(None).value
         else:
             value = self.value
-        result = (unit * rounder(value / float(unit)).astype('i8'))
+        if time_unit == 'N':
```

Comment:
Ah I see, this method makes more sense. Thanks!
 
I mistakenly thought `.nanos` returned the numeric value after the time unit instead of the frequency in nanoseconds which is why I was using this method.